### ISSUE TITLE
Use formatters on field level in @Table POJOs

### ISSUE THREAD

**Issue by nikowitt** [TNG/JGiven#252]
Hi Jan,

I currently have to implement test scenarios that heavily rely on the @Table annotation.

```

	private static final Code RATE_11_TO_20 = Code.of(ScaffoldCode.class, 11, 20, 16, 8, 0.0171);
	private static final Code RATE_31_PLUS = Code.of(ScaffoldCode.class, 31, 1000, 13, 6.5, 0.0171);

	@Test
	public void first_scenario() {
		given().codes_with_rates(RATE_11_TO_20, RATE_31_PLUS);
		(...)
	{

(...)

	public CostCalculationStage codes_with_rates(@Table Code... rates) {
		(...)
	}

```

where Code is
```
	static class Code {

		private final Class<? extends BaseCode<?>> clazz;
		private final int minSize;
		private final int maxSize;
		private final double erection;
		private final double dismantling;
		private final double rent;

		private Code(Class<? extends BaseCode<?>> clazz, int minSize, int maxSize, double erection, double dismantling, double rent) {
			this.clazz = clazz;
			this.erection = erection;
			this.dismantling = dismantling;
			this.rent = rent;
			this.minSize = minSize;
			this.maxSize = maxSize;

		}

		public static Code of(Class<? extends BaseCode<?>> clazz, int minSize, int maxSize, double erection, double dismantling, double rent) {
			return new Code(clazz, minSize, maxSize, erection, dismantling, rent);
		}

		public double getErection() {
			return erection;
		}

		public double getDismantling() {
			return dismantling;
		}

		public double getRent() {
			return rent;
		}

		public int getMinSize() {
			return minSize;
		}

		public int getMaxSize() {
			return maxSize;
		}

		public Class<? extends BaseCode<?>> getClazz() {
			return clazz;
		}

	}
```

creates the output:

```
   Given codes with rates

     | clazz                                       | minSize | maxSize | erection | dismantling |   rent |
     +---------------------------------------------+---------+---------+----------+-------------+--------+
     | class com.sobis.sms.model.code.ScaffoldCode |      11 |      20 |     16.0 |         8.0 | 0.0171 |
     | class com.sobis.sms.model.code.ScaffoldCode |      31 |    1000 |     13.0 |         6.5 | 0.0171 |
```

It would be really cool if in this case, formatter annotations could also be recognized on field level, e.g.

```
static class Code {
(...)
@SimpleClassName
private final Class<? extends BaseCode<?>> clazz;
(...)
}
```
to create an output like

```
     | clazz        | minSize | maxSize | erection | dismantling |   rent |
     +--------------+---------+---------+----------+-------------+--------+
     | ScaffoldCode |      11 |      20 |     16.0 |         8.0 | 0.0171 |
     | ScaffoldCode |      31 |    1000 |     13.0 |         6.5 | 0.0171 |
```

What do you think?

Best regards,
Niko

**Comment by janschaefer** [comment]
Makes totally sense to me :-)

**Comment by janschaefer** [comment]
Btw. did you have a look at https://github.com/TNG/JGiven/issues/122?

**Comment by nikowitt** [comment]
No, I forgot that this ability exist :)
But generally speaking it would be less verbose if formatters could be applied directly.

**Comment by janschaefer** [comment]
Yes. I agree. There should be the possibility to have formatters at field level.

**Comment by janschaefer** [comment]
With PR #297 it will be easy to support this feature.

**Comment by janschaefer** [comment]
Fixed with #297
